include/actionlib/server/simple_action_server_imp.h

```diff
@@ -153,6 +153,11 @@ namespace actionlib {
       }
 
       assert(execute_thread_);
+      if (!execute_thread_)
+      {
+        ROS_ERROR_NAMED("actionlib", "execute_thread_ is null, skipping attempt to join it");
+        return;
+      }
```

Comment:
This is a case which shouldn't be possible based on the logic of the class. An assert which is also effective in release builds would be sufficient imo.

It could also be changed to:

```
if (execute_thread_) {
  execute_thread_->join();
  delete execute_thread_;
  execute_thread_ = NULL;
}
```